### PR TITLE
Metrics default val set 0

### PR DESCRIPTION
**Pull request by wxbty**
## What is the purpose of the change
1、For metrics data sets at the servce and method levels, if a key exists in a certain category of data, it should set the default 0 for other categories to make the data aligned
2、some code opt/ add metrics code comment。delete MetricsKey.formatName,It will change the attribute value of the enum causing ci to fail. 
3、MetricSample.Type changed from invocation to built-in property of MetricsKeyWrapper to avoid fillZero injecting a fixed key

## Verifying this change
add testcase.

# TYPE dubbo_registry_subscribe_service_num_total gauge
dubbo_registry_subscribe_service_num_total{application_module_id="1.1",application_name="dubbo-springboot-demo-provider",application_version="3.2.2-SNAPSHOT",git_commit_id="9df27cb8b256fed00928a4958e1977d6b51d1bcc",hostname="linyaoMac.local",interface="dubbo-springboot-demo-provider/org.apache.dubbo.metrics.service.MetricsService:1.0.0",ip="10.201.94.160",} 0.0
dubbo_registry_subscribe_service_num_total{application_module_id="1.1",application_name="dubbo-springboot-demo-provider",application_version="3.2.2-SNAPSHOT",git_commit_id="9df27cb8b256fed00928a4958e1977d6b51d1bcc",hostname="linyaoMac.local",interface="dubbo-springboot-demo-provider/org.apache.dubbo.metadata.MetadataService:1.0.0",ip="10.201.94.160",} 0.0
dubbo_registry_subscribe_service_num_total{application_module_id="1.1",application_name="dubbo-springboot-demo-provider",application_version="3.2.2-SNAPSHOT",git_commit_id="9df27cb8b256fed00928a4958e1977d6b51d1bcc",hostname="linyaoMac.local",interface="org.apache.dubbo.springboot.demo.DemoService",ip="10.201.94.160",} 0.0

### CODE REVIEW

**Comment by AlbumenJ** [comment]
image

Please fix failed test cases

**Comment by wxbty** [comment]
> image
> 
> Please fix failed test cases

done

**Review by AlbumenJ — APPROVED** [review]
LGTM

**Review by AlbumenJ — APPROVED** [review]
LGTM

**Comment by AlbumenJ** [comment]
@songxiaosheng PTAL

File: dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/data/MethodStatComposite.java [hunk 1]
```diff
@@ -58,18 +65,21 @@ public void incrementMethodKey(MetricsKeyWrapper wrapper, Invocation invocation,
             return;
         }
         methodNumStats.get(wrapper).computeIfAbsent(new MethodMetric(getApplicationModel(), invocation), k -> new AtomicLong(0L)).getAndAdd(size);
```
Comment (songxiaosheng) [inline dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/data/MethodStatComposite.java:67]: 初始化0操作关于接口运行时产生的指标是否可以放到服务初始化那里，我们初始化完服务立即初始化当前接口的相关指标，这样RPC请求的数据可以提前初始化为0。

服务初始化的逻辑如下位置，DefaultModuleDeployer类型
 // export services
  exportServices();

 // refer services
 referServices();

**Pull request merged**